### PR TITLE
Menubar

### PR DESCRIPTION
**Pull request by kwendel**
UI now has a menubar and mouse gives a loading icon when the model is loading.
Also did a refactor on the MVC pattern based on the TA comments, the view now makes the controller

### CODE REVIEW

**Comment by kwendel** [comment]
Still needs a rebase with master...

File: src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java [hunk 1]
```diff
@@ -15,26 +17,50 @@
 import java.util.List;
 import java.util.Map;
 
-public class GraphDataRepository extends AbstractGraphData {
-	private int longestNodePath;
-	
+public class GraphDataRepository extends AbstractGraphData implements LoadingObservable {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java]: I think it's better to split up the GraphDataRepository and put its parsing methods into a separate Parser class.

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 2]
```diff
@@ -33,124 +41,160 @@
...
+			if (confirm("Error!", "Your file was not found. Want to try again?")) {
+				makeGraph();
+			}
+		}
+	}
+
+	/**
+	 * Parses the phylogentic tree and makes a sidebarview
+	 */
+	public void makePhyloTree() {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: I think you want to handle this stuff in the dedicated controller and not in the Application class. That's whats controllers are for.

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 3]
```diff
@@ -33,124 +41,160 @@
...
 	}
-	
+
 	/**
-	 * Get the views and set them to the application Set the key and window
-	 * controllers.
+	 * Parses the graph files and makes a graphview.
 	 */
-	public void start() {
+	public void makeGraph() {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: I think you want to handle this stuff in the dedicated controller and not in the Application class. That's whats controllers are for.

**Comment by kwendel** [comment]
I agree on that, lets put in our next sprint.

File: src/main/java/tudelft/ti2806/pl3/Controller.java [hunk 4]
```diff
@@ -9,6 +7,5 @@
  * Created by Boris Mattijssen on 30-04-15.
  */
 public interface Controller {
-	
-	Component getPanel();
+
 }
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/Controller.java]: Empty interface, remove it?

**Comment by Delfil** [comment]
There are still some javadocs missing, but everything else seems fine.

File: src/main/java/tudelft/ti2806/pl3/View.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Interface for all the views used in the application.
+ * Created by Kasper on 27-5-2015.
+ */
+public interface View {
+
+	Component getPanel();
+
+	Controller getController();
```
Comment (kwendel) [inline src/main/java/tudelft/ti2806/pl3/View.java]: Controller interface should not be removed, it is still used in the getter of the controller in the views

Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/View.java]: So, you are casting it after using the getController method?

**Comment by kwendel** [comment]
branch is rebased with the master on menubarRebase2

**Comment by Delfil** [comment]
Already merged with menubarRebase2

**Pull request closed without merging**